test/core/event_engine/iomgr_event_engine/timer_heap_test.cc

```diff
@@ -0,0 +1,209 @@
+/*
+ *
+ * Copyright 2015 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+#include "src/core/lib/event_engine/iomgr_engine/timer_heap.h"
+
+#include <stdlib.h>
+#include <string.h>
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "src/core/lib/event_engine/iomgr_engine/timer.h"
+#include "src/core/lib/gpr/useful.h"
+#include "src/core/lib/gprpp/bitset.h"
+
+using testing::Contains;
+using testing::Not;
+
+namespace grpc_event_engine {
+namespace iomgr_engine {
+
+namespace {
+int64_t RandomDeadline(void) { return rand(); }
+
+std::vector<Timer> CreateTestElements(size_t num_elements) {
+  std::vector<Timer> elems(num_elements);
+  for (size_t i = 0; i < num_elements; i++) {
+    elems[i].deadline = RandomDeadline();
+  }
+  return elems;
+}
+}  // namespace
+
+static void CheckValid(TimerHeap* pq) {
+  const std::vector<Timer*>& timers = pq->TestOnlyGetTimers();
+  for (size_t i = 0; i < timers.size(); ++i) {
+    size_t left_child = 1u + 2u * i;
+    size_t right_child = left_child + 1u;
+    if (left_child < timers.size()) {
+      EXPECT_LE(timers[i]->deadline, timers[left_child]->deadline);
+    }
+    if (right_child < timers.size()) {
+      EXPECT_LE(timers[i]->deadline, timers[right_child]->deadline);
+    }
+  }
+}
+
+/*******************************************************************************
+ * test1
+ */
+
+TEST(TimerHeapTest, Test1) {
```

Comment:
Can we give these tests better names than `Test1` and `Test2`?

Reply:
done